src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -269,8 +269,16 @@ private static void AddSolutionFolders(this SlnFile slnFile, SlnProject slnProje
                 while (nestedProjects.ContainsKey(id))
                 {
                     id = nestedProjects[id];
-                    var parentSlnProject = solutionFolderProjects.Where(p => p.Id == id).Single();
-                    path = Path.Combine(parentSlnProject.FilePath, path);
+                    try
+                    {
+                        var parentSlnProject = solutionFolderProjects.Where(p => p.Id == id).Single();
+                        path = Path.Combine(parentSlnProject.FilePath, path);
+                    }
+                    catch (System.InvalidOperationException)
+                    {
+                        Reporter.Error.WriteLine(String.Format(CommonLocalizableStrings.CorruptSolutionProjectFolderStructure, slnFile.FullPath, id).Red());
+                        Environment.Exit(1);
+                    }
```

Comment:
Rather than catching the exception, it would be better to check for the error.

Also, I think throwing a GracefulException is the way we usually handle errors like this where they need to be displayed and the program needs to exit.

```suggestion
                    try
                    {
                    var parentSlnProject = solutionFolderProjects.Where(p => p.Id == id).SingleOrDefault();
                    if (parentSlnProject == null)
                    {
                        throw new GracefulException(CommonLocalizableStrings.CorruptSolutionProjectFolderStructure, slnFile.FullPath);
                    }
                    path = Path.Combine(parentSlnProject.FilePath, path);
```